luxonis_train/nodes/backbones/ddrnet/ddrnet.py

```diff
@@ -0,0 +1,298 @@
+"""DDRNet backbone.
+
+Adapted from: U{https://github.com/Deci-AI/super-gradients/blob/master/src/super_gradients/training/models/segmentation_models/ddrnet.py}
+Original source: U{https://github.com/ydhongHIT/DDRNet}
+Paper: U{https://arxiv.org/pdf/2101.06085.pdf}
+@license: U{https://github.com/Deci-AI/super-gradients/blob/master/LICENSE.md}
+"""
+from typing import Dict, Type
+
+from torch import Tensor, nn
+
+from luxonis_train.nodes.base_node import BaseNode
+from luxonis_train.nodes.blocks import ConvModule
+
+from .blocks import (
+    DAPPM,
+    BasicDDRBackBone,
+    BasicResNetBlock,
+    Bottleneck,
+    UpscaleOnline,
+    _make_layer,
+)
+
+
+class DDRNet(BaseNode[Tensor, list[Tensor]]):
+    def __init__(
+        self,
+        use_aux_heads: bool = True,
+        upscale_module: nn.Module = None,
+        highres_planes: int = 64,
+        spp_width: int = 128,
+        ssp_inter_mode: str = "bilinear",
+        segmentation_inter_mode: str = "bilinear",
+        block: Type[nn.Module] = BasicResNetBlock,
+        skip_block: Type[nn.Module] = BasicResNetBlock,
+        layer5_block: Type[nn.Module] = Bottleneck,
+        layer5_bottleneck_expansion: int = 2,
+        spp_kernel_sizes: list[int] = None,
+        spp_strides: list[int] = None,
+        layer3_repeats: int = 1,
+        planes: int = 32,
+        layers: list[int] = None,
+        input_channels: int = 3,
```

Comment:
There's already a property in `BaseNode` called `in_channels` that gives you the right number of input channels. It doesn't (shouldn't) have to be specified by the user in parameters.